### PR TITLE
Making liveliness tests more stable on Windows [5872]

### PR DESCRIPTION
**Pull request by raquelalvarezbanos**

### CODE REVIEW

**Comment by raquelalvarezbanos** [comment]
Jenkins builds:

Platform | Pass | Fail | Iterations
--- | --- | --- | ---
Windows | [#20](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows%20Test/20/) | [#22](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows%20Test/22/) (unrelated) | 2
Linux | [#15](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20Test/15/) | | 1

**Comment by raquelalvarezbanos** [comment]
Tests on the different platforms passed ([Mac](http://ci.intranet.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/28/console), [Windows](http://ci.intranet.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/21/), [Linux](http://ci.intranet.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/24/)) although overall result is marked as failure (http://ci.intranet.eprosima.com:8080/job/FastRTPS%20Manual/24/console)

File: test/blackbox/BlackboxTestsLivelinessQos.cpp [hunk 1]
```diff
@@ -134,13 +134,13 @@ TEST(LivelinessQos, ShortLiveliness_ManualByParticipant_Reliable)
...
+    unsigned int count = 0;
     for (auto data_sample : data)
     {
         writer.send_sample(data_sample);
-        ++count;
-        reader.block_for_at_least(count);
-        std::this_thread::sleep_for(std::chrono::milliseconds(writer_sleep_ms));
+        reader.wait_liveliness_recovered(count + 1);
+        reader.wait_liveliness_lost(count + 1);
+        count++;
```
Comment (LuisGP) [inline test/blackbox/BlackboxTestsLivelinessQos.cpp]: Why not keep `++count;` at the beginning and call the methods just with `count`?

``` cpp
++count;
reader.wait_liveliness_recovered(count);
reader.wait_liveliness_lost(count);
```

File: test/blackbox/BlackboxTestsLivelinessQos.cpp [hunk 2]
```diff
@@ -195,13 +196,13 @@ TEST(LivelinessQos, ShortLiveliness_ManualByParticipant_BestEffort)
...
```
Comment (LuisGP) [inline test/blackbox/BlackboxTestsLivelinessQos.cpp]: Same as above.

File: test/blackbox/BlackboxTestsLivelinessQos.cpp [hunk 3]
```diff
@@ -670,13 +672,13 @@ TEST(LivelinessQos, ShortLiveliness_ManualByParticipant_Automatic_Reliable)
...
```
Comment (LuisGP) [inline test/blackbox/BlackboxTestsLivelinessQos.cpp]: Again the same comment.

File: test/blackbox/BlackboxTestsLivelinessQos.cpp [hunk 4]
```diff
@@ -786,13 +789,13 @@ TEST(LivelinessQos, ShortLiveliness_ManualByParticipant_Automatic_BestEffort)
...
```
Comment (LuisGP) [inline test/blackbox/BlackboxTestsLivelinessQos.cpp]: +1

**Comment by richiware** [comment]
Build status:
                                * Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/${linux_number}/)
                                * Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/${windows_number}/)
                                * Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/${mac_number}/)

File: test/blackbox/BlackboxTestsLivelinessQos.cpp (hunk 1 above)
Comment (raquelalvarezbanos) [reply test/blackbox/BlackboxTestsLivelinessQos.cpp]: :+1: Changed

**Comment by raquelalvarezbanos** [comment]
The overall result failed but tests on each platform passed:
[Mac](http://ci.intranet.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/41/)
[Windows](http://ci.intranet.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/34/)
[Linux](http://ci.intranet.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/37/)

**Pull request merged**